sphinx/source/docs/user_guide/source_examples/extensions_example_widget.py

```diff
@@ -0,0 +1,416 @@
+    # can be found here:
+    #
+    #    http://bokeh.pydata.org/en/latest/docs/reference/core.html#bokeh-core-properties
+    disable = Bool(default=True, help="""
+    Enable or disable the slider.
+    """)
+
+    range = Tuple(Float, Float, default=(0.1, 0.9), help="""
+    Initial or selected range.
+    """)
+
+    start = Float(default=0, help="""
+    The minimum allowable value.
+    """)
+
+    end = Float(default=1, help="""
+    The maximum allowable value.
+    """)
+
+    step = Float(default=0.1, help="""
+    The step between consecutive values.
+    """)
+
+    orientation = Enum("horizontal", "vertical", help="""
+    Orient the slider either horizontally (default) or vertically.
+    """)
+
+    callback = Instance(Callback, help="""
+    A callback to run in the browser whenever the current Slider value changes.
+    """)
+
+    callback_throttle = Float(default=200, help="""
+    Number of microseconds to pause between callback calls as the slider is moved.
+    """)
+
+    callback_policy = Enum(SliderCallbackPolicy, default="throttle", help="""
+    When the callback is initiated. This parameter can take on only one of three options:
+       "continuous": the callback will be executed immediately for each movement of the slider
+       "throttle": the callback will be executed at most every ``callback_throttle`` milliseconds.
+       "mouseup": the callback will be executed only once when the slider is released.
+       The `mouseup` policy is intended for scenarios in which the callback is expensive in time.
+    """)
+
+
+x = [x*0.005 for x in range(2, 198)]
+y = x
+w = [w*0.005 for w in range(2, 198)]
+z = [z**0.5 for z in w]
+
+source = ColumnDataSource(data=dict(x=x, y=y, w=w, z=z))
+
+plot = Figure(plot_width=400, plot_height=400)
+plot.line('x', 'y', source=source, line_width=3, line_alpha=0.6, color='#ed5565')
+plot.line('w', 'z', source=source, line_width=3, line_alpha=0.6)
+
+callback_single = CustomJS(args=dict(source=source), code="""
+        var data = source.get('data');
+        var f = cb_obj.get('value')
+        x = data['x']
+        y = data['y']
+        w = data['w']
+        z = data['z']
+        for (i = 0; i < x.length; i++) {
+            y[i] = Math.pow(x[i], f)
+            z[i] = Math.pow(w[i], f/2)
+        }
+        source.trigger('change');
+    """)
+
+
+callback_ion = CustomJS(args=dict(source=source), code="""
+        var data = source.get('data');
+        var f = cb_obj.get('range')
+        x = data['x']
+        y = data['y']
+        pow = (Math.log(y[100])/Math.log(x[100]))
+        console.log(pow)
+        delta = (f[1]-f[0])/x.length
+        for (i = 0; i < x.length; i++) {
+            x[i] = delta*i + f[0]
+            y[i] = Math.pow(x[i], pow)
+        }
+        source.trigger('change');
+    """)
+
+callback_jquery = CustomJS(args=dict(source=source), code="""
+        var data = source.get('data');
+        var f = cb_obj.get('range')
```

Comment:
> as in cb_obj.attr('range') etc?

@djpugh, you mean `cb_obj.get('range')`?. Yes, it will be simply `cb_obj.range`.